pkg/skaffold/build/local/local.go

```diff
@@ -151,3 +151,24 @@ func (b *Builder) getImageIDForTag(ctx context.Context, tag string) (string, err
 	}
 	return insp.ID, nil
 }
+
+func (b *Builder) SyncMap(ctx context.Context, a *latest.Artifact) (map[string][]string, error) {
+	if a.DockerArtifact == nil {
+		return nil, build.ErrSyncMapNotSupported{}
+	}
+
+	syncMap, err := docker.SyncMap(ctx, a.Workspace, a.DockerArtifact.DockerfilePath, a.DockerArtifact.BuildArgs, b.insecureRegistries)
+
+	if err != nil {
+		// if the context was cancelled act as if all is well
+		// TODO(dgageot): this should be even higher in the call chain.
+		if ctx.Err() == context.Canceled {
```

Comment:
Can we fix that first to simplify the code here?

Reply:
I moved the checks to the outermost call site. Please check carefully :)